### PR TITLE
add F5 BigIP

### PR DESCRIPTION
**Pull request by Yofial**
F5 BigIP is a reverse proxy divided into separate modules for load balancing, web application firewall and more.

I've found these public examples from [shodan](https://www.shodan.io/):
1. https://weatherford.com/
2. https://kbr.com/

Many other detections from shodan are login pages to services I was not sure it is appropriate to publish here.

An indication of the size of the user base can be found [at this post](https://community.spiceworks.com/topic/2149576-receive-a-complimentary-forrester-wave-report-for-leading-waf-offerings). You can see the size of F5 in the Forrester Wave. The size indicates the market presence. This report refers to the Web Application Firewall which is only one module of F5 BigIP.

### CODE REVIEW

File: src/apps.json [hunk 1]
```diff
@@ -3098,6 +3098,26 @@
       "script": "ext-base\\.js",
       "website": "https://www.sencha.com"
     },
+    "F5 BigIP": {
+      "cats": [
+        64
+      ],
+      "headers": {
+        "server": "BigIP|BIG-IP"
```
Comment (jvoisin) [inline src/apps.json]: What about something like `"server": "^big-?ip$"` instead?

Comment (Yofial) [reply src/apps.json]: Good point. Fixing...

**Comment by Yofial** [comment]
Reverse proxy priority was changed for debug purposes.
Fixing it now

**Comment by HazCod** [comment]
An important one is that whenever a F5 firewall is enabled, cookies are set that start with `TSxxxx`.
https://support.f5.com/csp/article/K6850

**Comment by jvoisin** [comment]
We don't have a nice way to match on this unfortunately.
Also, it'll likely be too false-positive prone.

**Comment by Yofial** [comment]
Regex of header and cookie names are is not currently supported. I chose to add the detection with what is available now.

The suggestion of @HazCod for detecting the F5 ASM module (Web application firewall) can be implemented without the risk of false positive if we had support for regex in cookie names. Take a look at the regex in the wafw00f project [here](https://github.com/EnableSecurity/wafw00f/blob/master/wafw00f/plugins/f5bigipasm.py).

In the same way, the F5 LTM (Loadbalancer) can be detected with a cookie starting with '^BIGipServer'. [docs](https://support.f5.com/csp/article/K6917)

**Comment by AliasIO** [comment]
Thanks!

**Pull request merged**